### PR TITLE
dankomax: solved issues with waits in PO

### PR DESCRIPTION
**Pull request by dankomax**

### CODE REVIEW

File: src/main/java/com/softserveinc/ita/rozetka/ProductPage.java [hunk 1]
```diff
@@ -3,23 +3,33 @@
...
+    }
+
+    public long getPrice() {
+        return getLong("//p[contains(@class, 'product-prices__big')]");
     }
 
     public ShoppingCartModal addToCart() {
-        $x("//ul[@class='product-buttons']//button[contains(@class,'buy-button')]").click();
+        $x("//ul[@class='product-buttons']//button[contains(@class,'buy-button')]")
+                .shouldBe(visible)
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/ProductPage.java:21]: why is this needed?

File: src/main/java/com/softserveinc/ita/rozetka/SearchResultsPage.java [hunk 2]
```diff
@@ -32,14 +28,21 @@ public SearchResultsPage resetFilters() {
...
+    public Product getProduct(String number) {
+        return new Product(number);
+    }
+
     public SearchResultsPage sortBy(ProductSort sort) {
+        String firstResultXpath = "(//div[contains(@class, 'goods-tile ')])[1]";
+        String firstResultText = getText(firstResultXpath);
         $x("//rz-sort//select").click();
         $x(String.format("//rz-sort//select//option[contains(@value, '%s')]", sort.getOptionXpath())).click();
+        $x(firstResultXpath).shouldNotBe(Condition.text(firstResultText));
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/SearchResultsPage.java:45]: shouldNotHave

File: src/main/java/com/softserveinc/ita/rozetka/modals/ShoppingCartModal.java [hunk 3]
```diff
@@ -1,20 +1,41 @@
...
+import java.util.List;
 
 public class ShoppingCartModal {
     public boolean isShoppingCartEmpty() {
         return isVisible("//div[@data-testid='empty-cart']");
     }
 
+    public ShoppingCartModal clear() {
+        List<SelenideElement> cartItems = $$x("//button[contains(@id, 'cartProductActions')]")
+                .shouldBe(sizeGreaterThanOrEqual(1));
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/modals/ShoppingCartModal.java:24]: if cart if empty, this will fail

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/ProductPage.java (hunk 1 above)
Comment (dankomax) [reply src/main/java/com/softserveinc/ita/rozetka/ProductPage.java]: removed shouldBe(visible)

File: src/main/java/com/softserveinc/ita/rozetka/SearchResultsPage.java (hunk 2 above)
Comment (dankomax) [reply src/main/java/com/softserveinc/ita/rozetka/SearchResultsPage.java]: done

File: src/main/java/com/softserveinc/ita/rozetka/modals/ShoppingCartModal.java (hunk 3 above)
Comment (dankomax) [reply src/main/java/com/softserveinc/ita/rozetka/modals/ShoppingCartModal.java]: added check for cartItemActionButton presence before loop

File: src/main/java/com/softserveinc/ita/rozetka/modals/ShoppingCartModal.java [hunk 4]
```diff
@@ -1,20 +1,44 @@
...
 
+    public ShoppingCartModal clear() {
+        String cartItemActionButtonXpath = "//button[contains(@id, 'cartProductActions')]";
+        if (isVisible(cartItemActionButtonXpath)) {
+            List<SelenideElement> cartItems = $$x(cartItemActionButtonXpath)
+                    .shouldBe(sizeGreaterThanOrEqual(1));
+            for (SelenideElement item : cartItems) {
+                item.shouldBe(visible).click();
+                $x("//div[contains(@id, 'cartProductActions')]//button")
+                        .shouldBe(visible)
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/modals/ShoppingCartModal.java:30]: drop shouldBe() for all click()

**Review by toleksyn — CHANGES_REQUESTED** [review]
